### PR TITLE
Updated MariaDB

### PR DESCRIPTION
**Pull request by WestleyK**
Not very much to change here.

### CODE REVIEW

File: database/mariadb/mariadb.def [hunk 1]
```diff
@@ -3,7 +3,7 @@ From: mariadb:10.3.9
 
 %post
 # <YOUR_USERNAME> is the user who will be executing the container,
-# just run: `whoami` and thats your username.
+# just run: `whoami` and that's your username.
 # eg. sed -ie "s/^#user.*/user = westleyk/" /etc/mysql/my.cnf
```
Comment (jchavezlavalle) [inline database/mariadb/mariadb.def]: Can we export an environment variable called `YOUR_USERNAME` with the username right before running `sed -ie "s/^#user.*/user = <YOUR_USERNAME>/" /etc/mysql/my.cnf` ?

**Review by jchavezlavalle — CHANGES_REQUESTED** [review]

File: database/mariadb/mariadb.def (hunk 1 above)
Comment (WestleyK) [reply database/mariadb/mariadb.def]: Good idea! :+1:

File: database/mariadb/README.md [hunk 2]
```diff
@@ -171,18 +190,20 @@ Then, exit the container:
...
 ```
 $ singularity instance start \
- -B mariadb/db:/var/lib/mysql \
- -B mariadb/log:/var/log/mysql \
- -B mariadb/run:/var/run/mysqld \
- mariadb.sif mariadb
+-B mariadb/db:/var/lib/mysql \
+-B mariadb/log:/var/log/mysql \
+-B mariadb/run:/var/run/mysqld \
+-B mariadb/etc/:/etc/mysql/mariadb.conf.d \
```
Comment (jmstover) [inline database/mariadb/README.md]:
```
-B mariadb/etc/:/etc/mysql/mariadb.conf.d \
-B mariadb/conf/:/etc/mysql/conf.d \
```
Is this really needed? I don't require binding the directories under /etc/mysql to run.

**Review by jmstover — CHANGES_REQUESTED** [review]

File: database/mariadb/README.md [hunk 3]
```diff
@@ -55,18 +55,35 @@ $ wget https://raw.githubusercontent.com/sylabs/examples/master/database/mariadb
 
 <br>
 
+**REMEMBER** to change `<YOUR_USERNAME>` to your username, otherwise you will get this error when `> mysql_install_db`:
+
+```
+chown: invalid user: '<YOUR_USER_NAME>'
```
Comment (jchavezlavalle) [inline database/mariadb/README.md]: This should be: `chown: invalid user: <YOUR_USERNAME>`

**Review by jchavezlavalle — CHANGES_REQUESTED** [review]

File: database/mariadb/mariadb.def [hunk 4]
```diff
@@ -2,10 +2,10 @@ Bootstrap: docker
 From: mariadb:10.3.9
 
 %post
-# <YOUR_USERNAME> is the user who will be executing the container,
-# just run: `whoami` and thats your username.
-# eg. sed -ie "s/^#user.*/user = westleyk/" /etc/mysql/my.cnf
-sed -ie "s/^#user.*/user = <YOUR_USERNAME>/" /etc/mysql/my.cnf
+# replace `your-name` with your username, run `whoami` to see your username
+YOUR_USERNAME="your-name"
+
```
Comment (jchavezlavalle) [inline database/mariadb/mariadb.def:7]: `YOUR_USERNAME` variable needs to be exported after its definition: `export YOUR_USERNAME`

**Review by jchavezlavalle — CHANGES_REQUESTED** [review]

File: database/mariadb/README.md [hunk 5]
```diff
@@ -34,10 +34,10 @@ Bootstrap: docker
 From: mariadb:10.3.9
 
 %post
-# <YOUR_USERNAME> is the user who will be executing the container,
-# just run: `whoami` and that's your username.
-# eg. sed -ie "s/^#user.*/user = westleyk/" /etc/mysql/my.cnf
-sed -ie "s/^#user.*/user = <YOUR_USERNAME>/" /etc/mysql/my.cnf
+# replace `your-name` with your username, run `whoami` to see your username
+YOUR_USERNAME="your-name"
+
```
Comment (jchavezlavalle) [inline database/mariadb/README.md:39]: Also export the username variable in here

**Review by jchavezlavalle — CHANGES_REQUESTED** [review]

File: database/mariadb/README.md (hunk 5 above)
Comment (jmstover) [reply database/mariadb/README.md:39]: It's only used in the file ... there isn't a need to export it

File: database/mariadb/mariadb.def (hunk 4 above)
Comment (jmstover) [reply database/mariadb/mariadb.def:7]: Same ... it's only used in that section so no need to export

File: database/mariadb/README.md (hunk 5 above)
Comment (WestleyK) [reply database/mariadb/README.md:39]: Yes, No, Maybe...

Comment (WestleyK) [reply database/mariadb/README.md:39]: `export`, or no `export`?

File: database/mariadb/README.md [hunk 6]
```diff
@@ -1,12 +1,12 @@
...
 
 In this example, we will run a simple database server.
 
 
 #### What you need:
  - Singularity, which you can download and install from [here](https://github.com/sylabs/singularity).
  - A text editor, like: `micro`, `vim` or `nano`.
- - Root access.
+ - A access token form [here](https://cloud.sylabs.io/auth) (for remote builder), or root access.
  - mySQL, installed by `sudo apt-get install mysql-server`.
```
Comment (jchavezlavalle) [inline database/mariadb/README.md]: uppercase first letter -> `MySQL`

**Review by jchavezlavalle — CHANGES_REQUESTED** [review]

**Review by jchavezlavalle — APPROVED** [review]

File: database/mariadb/README.md [hunk 7]
```diff
@@ -92,9 +105,41 @@ Once we are in the container, setup MariaDB:
...
+>  mysql_install_db
+/usr/sbin/mysqld: Can't read dir of '/etc/mysql/mariadb.conf.d/' (Errcode: 13 "Permission denied")
+/usr/sbin/mysqld: Can't read dir of '/etc/mysql/conf.d/' (Errcode: 13 "Permission denied")
+Fatal error in defaults handling. Program aborted
+[...]
+```
+
+<br>
+
+Then you need to clear `/etc/apparmor.d/usr.sbin.mysqld`:<br>
```
Comment (jmstover) [inline database/mariadb/README.md]: Change to:
```
Then you may need to clear `/etc/apparmor.d/usr.sbin.mysqld`:<br>
```
Add "may" to it.

**Review by jmstover — CHANGES_REQUESTED** [review]

**Review by jmstover — APPROVED** [review]

**Pull request merged**